### PR TITLE
document the bindings massacre

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -34,6 +34,8 @@ Changelog
 * Added a check to :class:`~cryptography.x509.NameConstraints` to ensure that
   :class:`~cryptography.x509.DNSName` constraints do not contain any ``*``
   wildcards.
+* Removed many unused CFFI OpenSSL bindings. This will not impact you unless
+  you are using ``cryptography`` to directly invoke OpenSSL's C API.
```
Comment (reaperhulk) [inline CHANGELOG.rst:38]: I feel like we should mention that our bindings aren't public API 😄

Comment (alex) [reply CHANGELOG.rst:38]: Meaning by not documenting these changes, or by including that note here?

Comment (reaperhulk) [reply CHANGELOG.rst:38]: Including the note here. I am in favor of saying we made changes but also that this isn’t supported surface area.

Comment (reaperhulk) [reply CHANGELOG.rst:38]: Maybe we say come talk about adding features to cryptography if you need some functionality you’re rummaging around in the bindings to get right now. 🤷‍♀️

Comment (alex) [reply CHANGELOG.rst]: Decided not to add the note, I don't want to be perceived as promising we'll add them, when we know some external users want to do things that are bad.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -34,6 +34,10 @@ Changelog
 * Added a check to :class:`~cryptography.x509.NameConstraints` to ensure that
   :class:`~cryptography.x509.DNSName` constraints do not contain any ``*``
   wildcards.
+* Removed many unused CFFI OpenSSL bindings. This will not impact you unless
+  you are using ``cryptography`` to directly invoke OpenSSL's C API. Note that
+  these have never been considered a stable, supported, public API by
+  ``cryptography``, this note is included purely as a curtesy.
```
Comment (reaperhulk) [inline CHANGELOG.rst:40]:
```suggestion
  ``cryptography``, this note is included as a courtesy.
```

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**